morphic.js

```diff
@@ -5213,7 +5253,7 @@ CursorMorph.prototype.gotoPos = function (aPoint) {
 
 CursorMorph.prototype.updateSelection = function (shift) {
     if (shift) {
-        if (!this.target.endMark && !this.target.startMark) {
+        if (isNil(this.target.endMark) && isNil(this.target.startMark)) {
```

Comment:
Exactly, and that was breaking selection behavior when you started selecting from the beginning of any text. You can test it out in the [live morphic](http://snap.berkeley.edu/snapsource/morphic.html) page. Just go to the beginning of any text and press shift + →.